pygen/pygen_src/riscv_instr_gen_config.py

```diff
@@ -194,6 +206,26 @@ def gpr_c(self):
                                            riscv_reg_t.ZERO, riscv_reg_t.RA, riscv_reg_t.GP))
         vsc.unique(self.gpr0, self.gpr1, self.gpr2, self.gpr3)
 
+    @vsc.constraint
+    def ra_c(self):
+        self.ra != riscv_reg_t.SP
+        self.ra != riscv_reg_t.TP
+        self.ra != riscv_reg_t.ZERO
+
+    @vsc.constraint
+    def reserve_scratch_reg_c(self):
```

Comment:
I actually recently updated this particular constraint, I think these updates need to be reflected here

Reply:
Sorry about that,
we were actually following a particular commit for the development. i.e [master](https://github.com/saurabhsingh-pvips/riscv-dv/tree/master) branch of our repo. 
anyways, I have updated the constraint with the latest one.